seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceReader.java

```diff
@@ -140,32 +172,19 @@ public void handleNoMoreSplits() {
 
     @Override
     public void notifyCheckpointComplete(long checkpointId) throws Exception {
-        if (this.metadata.isCommitOnCheckpoint()) {
-            consumer.commitSync();
-        }
-    }
-
-    private KafkaConsumer<byte[], byte[]> initConsumer(String bootstrapServer, String consumerGroup,
-                                                       Properties properties, boolean autoCommit) {
-        Properties props = new Properties();
-        properties.forEach((key, value) -> props.setProperty(String.valueOf(key), String.valueOf(value)));
-        props.setProperty(ConsumerConfig.GROUP_ID_CONFIG, consumerGroup);
-        props.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer);
-        props.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, CLIENT_ID_PREFIX + "-enumerator-consumer-" + this.hashCode());
-
-        props.setProperty(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
-                ByteArrayDeserializer.class.getName());
-        props.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
-                ByteArrayDeserializer.class.getName());
-        props.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, String.valueOf(autoCommit));
-
-        // Disable auto create topics feature
-        props.setProperty(ConsumerConfig.ALLOW_AUTO_CREATE_TOPICS_CONFIG, "false");
-        return new KafkaConsumer<>(props);
-    }
-
-    private Set<TopicPartition> convertToPartition(Collection<KafkaSourceSplit> sourceSplits) {
-        return sourceSplits.stream().map(KafkaSourceSplit::getTopicPartition).collect(Collectors.toSet());
+        consumerThreadMap.forEach((split, consumerThread) -> {
+            try {
+                consumerThread.getTasks().put(consumer -> {
+                    if (this.metadata.isCommitOnCheckpoint()) {
+                        Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
+                        offsets.put(split.getTopicPartition(), new OffsetAndMetadata(split.getEndOffset()));
```

Comment:
This has the same problem, because `pollNext#135` keeps calling `split#setEndOffset`;
You can backup the offset info when the checkpoint is triggered in the `#snapshotState`.